iproute2: 4.5.0 -> 4.7.0

groxxda opened the pull request:
###### Motivation for this change

New version available
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

iproute now packages a bash-completion file which it installs to
$BASH_COMPDIR.

groxxda committed:
iproute2: 4.5.0 -> 4.7.0

iproute now packages a bash-completion file which it installs to
$BASH_COMPDIR.

fpletz commented on pkgs/os-specific/linux/iproute/default.nix:
```diff
@@ -3,11 +3,12 @@
+    sha256 = "14kvpdlxq8204f5ayfdfb6mc0423mbf3px9q0spdly9sng7xnq4g";
   };
 
   patches = lib.optionals enableFan [
```
The fan patches won't apply. This breaks the `fanctl` package. Either those patches have to be fixed or add an old iproute with fan patches for `fanctl`.

fpletz commented:
Related issue: #17259

groxxda committed:
iproute2: fanpatch: adjust for new version

- The patch did not apply because the code around the additions changed.
- The patch uses functions that got changed [1] & [2], I adjusted the
  patch to use the safe version. Probably not needed but better safe
  than sorry.
[1] format_host: http://git.kernel.org/cgit/linux/kernel/git/shemminger/iproute2.git/commit/?id=a418e451643e77fe36861e53359587ba8aa41873
[2] rt_addr_n2a: http://git.kernel.org/cgit/linux/kernel/git/shemminger/iproute2.git/commit/?id=7faf1588a755edb9c9cabbe1d3211265e9826d28

groxxda commented:
@fpletz fixed the patches, see commit message

cc @cstrahan you seem to use the fan networking stuff, could you test this?

fpletz commented:
@groxxda When starting bash, I encounter the following error:

```bash: .: /run/current-system/sw/share/bash-completion/completions/iproute2: is a directory```

Can you take a look?

groxxda commented:
@fpletz i'll see what i can do

groxxda commented:
```
From a8e3f56c76152e0cdf62fbc903525ac38c649050 Mon Sep 17 00:00:00 2001
From: Alexander Ried <ried@mytum.de>
Date: Tue, 27 Sep 2016 17:49:03 +0200
Subject: [PATCH] iproute2: fix bash completion

apparently bash expects only files in its completion folder and not
subfolders.
---
 pkgs/os-specific/linux/iproute/default.nix | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkgs/os-specific/linux/iproute/default.nix b/pkgs/os-specific/linux/iproute/default.nix
index 9d3d76e..6f6c0c6 100644
--- a/pkgs/os-specific/linux/iproute/default.nix
+++ b/pkgs/os-specific/linux/iproute/default.nix
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
     "LIBDIR=$(out)/lib"
     "SBINDIR=$(out)/sbin"
     "MANDIR=$(out)/share/man"
-    "BASH_COMPDIR=$(out)/share/bash-completion/completions/iproute2"
+    "BASH_COMPDIR=$(out)/share/bash-completion/completions"
     "DOCDIR=$(TMPDIR)/share/doc/${name}" # Don't install docs
   ];
 
-- 
2.10.0
```

should fix it.

fpletz commented:
Thanks! Pushed as d666196a44503ae809f64266851152285803d755.
